rpm/caddy.spec

```diff
@@ -0,0 +1,134 @@
+# https://bugzilla.redhat.com/show_bug.cgi?id=995136#c12
+%global _dwz_low_mem_die_limit 0
+
+Name:           caddy
+# https://docs.fedoraproject.org/en-US/packaging-guidelines/Versioning/#_versioning_prereleases_with_tilde
+Version:        2.0.0~beta6
+%global tag     v2.0.0-beta6
+Release:        1%{?dist}
+Summary:        HTTP/2 web server with automatic HTTPS
+License:        ASL 2.0
+URL:            https://caddyserver.com
+
+# In order to build Caddy with version information, we need to import it as a
+# go module.  To do that, we are going to forgo the traditional source tarball
+# and instead use this simple go file.  This method requires that we allow
+# networking in the build environment.
+Source0:        main.go
+# Use official resources for config, unit file, and welcome page.
+# https://github.com/caddyserver/dist
+Source1:        https://raw.githubusercontent.com/caddyserver/dist/master/config/Caddyfile
+Source2:        https://raw.githubusercontent.com/caddyserver/dist/master/init/caddy.service
+Source3:        https://raw.githubusercontent.com/caddyserver/dist/master/welcome/index.html
+# Since we are not using a traditional source tarball, we need to explicitly
+# pull in the license file.
+Source4:        https://raw.githubusercontent.com/caddyserver/caddy/%{tag}/LICENSE
+
+BuildRequires:  golang >= 1.13
+BuildRequires:  git-core
+BuildRequires:  systemd
+%{?systemd_requires}
+Provides:       webserver
+
+
+%description
+Caddy is the HTTP/2 web server with automatic HTTPS.
+
+
+%prep
+%setup -q -c -T
+# Copy main.go and LICENSE into the build directory.
+cp %{S:0} %{S:4} .
+
+
+%build
+go mod init caddy
+echo "require github.com/caddyserver/caddy/v2 %{tag}" >> go.mod
+go build \
+    -buildmode pie \
+    -compiler gc \
```

Comment:
Any downside for including [`-trimpath`](https://golang.org/doc/go1.13#go-command) for build that doesn't expose local path of the build server and be more reproducible?

Reply:
I'm not familiar with go build flags to say for sure.  I got these flags from Fedora's `%gobuild` macro.  I didn't use the macro directly because it turns off go modules.  Are you saying that without this flag, build server paths are encoded (maybe the wrong term here) into the final binary?

Reply:
Yes. For example, on my laptop, Caddy lives in `/Users/mohammed/github/caddy`. If I build Caddy without it, then execute `strings {caddy_executable}`, you'll find lines listing the paths of Caddy source files prefixed with `/Users/mohammed/github/caddy`, e.g. `/Users/mohammed/github/caddy/admin.go` etc. It'll just be `github.com/caddyserver/caddy/admin.go`, or something like this.

> -trimpath
	remove all file system paths from the resulting executable.
	Instead of absolute file system paths, the recorded file names
	will begin with either "go" (for the standard library),
	or a module path@version (when using modules),
	or a plain import path (when using GOPATH).

https://golang.org/cmd/go/#hdr-Compile_packages_and_dependencies

Reply:
Yeah -- I think so anyway. Stack traces include (absolute) file paths without trimpath. Trimming the path can make stack traces a little more readable and the full file paths don't mean anything useful outside of dev environments.

Reply:
According to a blogpost by Filippo, passing `-s -w` to `-ldflags` also still leaves meaningful information in the stack trace and not hinder understanding the underlying cause. But, I don't know if there are any negative consequences. Linkers are beyond my knowledge level now.

Reply:
@Conan-Kudo Do you know if there is a reason Fedora's `%gobuild` doesn't include `-trimpath`?  I found a few hits for that in the mailing list for build log output, but my recent copr builds don't have it.

Reply:
I suspect that @jcajka didn't know about the flag? He's the one who maintains the build flags we set for `%gobuild`.

Reply:
@carlwgeorge @Conan-Kudo the flag had landed in Apr this year in what to become 1.13, so there hasn't been really a chance for adding it with all the stuff happening in the f31. It does seems to me sensible to add it from end user perspective, compiler maintainer's..., not sure how it will be handled by the "new macros" infrastructure. Could you please open a Fedora BZ against go-rpm-macros package in Fedora or feel free to open PR. Thanks for bringing this up.

Reply:
https://bugzilla.redhat.com/show_bug.cgi?id=1763889